roles/foreman_installer/tasks/install.yml

```diff
@@ -6,17 +6,19 @@
     msg: 'foreman_installer_options must be an array'
   when: foreman_installer_options is string
 
-- name: 'Join options'
+- name: Set disable system checks if present
   set_fact:
-    foreman_installer_options_joined: "{{ foreman_installer_options | join(' ') }} {{ foreman_installer_options_internal_use_only | join(' ') }}"
+    disable_system_checks: "{{ (foreman_installer_disable_system_checks|bool) | ternary(['--disable-system-checks'], []) }}"
 
-- name: 'Run installer'
-  shell: >
-      {{ foreman_installer_command }} {{ (foreman_installer_verbose|bool) | ternary("-v", "") }}
-      {{ (foreman_installer_no_colors|bool) | ternary("--no-colors", "") }}
-      {{ (foreman_installer_disable_system_checks|bool) | ternary("--disable-system-checks", "") }}
-      {{ foreman_installer_scenario_flag }} {{ foreman_installer_scenario }}
-      {{ foreman_installer_options_joined }}
+- name: Run installer
+  import_role:
+    name: theforeman.operations.installer
+  vars:
+    installer_scenario: "{{ foreman_installer_scenario }}"
+    installer_command: "{{ foreman_installer_command }}"
+    installer_verbose: "{{ foreman_installer_verbose|bool }}"
+    installer_no_colors: "{{ foreman_installer_no_colors|bool }}"
+    installer_options: "{{ foreman_installer_options + foreman_installer_options_internal_use_only + disable_system_checks }}"
```

Comment:
needs foreman prefix now, or rather dropping them here, as they are already defined in `foreman_installer/defaults`

(options one will be funny)